test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentTagHelperTest.cs

```diff
@@ -356,5 +415,25 @@ public Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
                 return Task.CompletedTask;
             }
         }
+
+        private static ViewContext CreateViewContext(HttpContext httpContext)
+        {
+            var serviceProvider = new ServiceCollection()
+                .AddSingleton<ITagHelperComponentPropertyActivator>(new TagHelperComponentPropertyActivator())
+                .BuildServiceProvider();
+            httpContext.RequestServices = serviceProvider;
+            var actionContext = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
```

Comment:
You don't care about a lot of the pieces in this method. Just add the bare minimum for `ViewContext` to function.